test/capture_udp.erl

```diff
@@ -74,6 +75,19 @@ peek() ->
 read() ->
     gen_server:call(?SERVER, read).
 
+-spec read_at_least(non_neg_integer()) -> {non_neg_integer(), iolist()}.
+read_at_least(Num) ->
```

Comment:
Just pushed a version that uses an accumulator.  I didn't go with a base case of 0 because it's possible that you'd get an N < 0 if you read more than you needed to.  Is this generally what you meant?